xcode/Subconscious/Shared/Components/Detail/UserProfileDetailMetaSheet.swift

```diff
@@ -0,0 +1,302 @@
+    var isFollowingUser: Bool
+    var send: (UserProfileDetailMetaSheetAction) -> Void
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack {
+                VStack(alignment: .leading, spacing: AppTheme.unit2) {
+                    if let user = profile.user {
+                        HStack {
+                            Text("@\(user.petname.verbatim)")
+                                .foregroundColor(.secondary)
+                            if user.category == .you {
+                            Text("(you)")
+                                .foregroundColor(.secondary)
+                            }
+                        }
+                        .font(.callout)
+                    }
+                }
+                Spacer()
+                CloseButtonView(action: { dismiss() })
+            }
+            .padding()
+            
+            Divider()
+            
+            ScrollView {
+                VStack(alignment: .center) {
+                    if let user = profile.user {
+                        MetaTableView {
+                            MetaTableItemShareLinkView(label: "Share DID", item: user.did.did)
+                            
+                            // None of these actions make sense to apply to yourself
+                            if user.category != .you {
+                                if isFollowingUser {
+                                    MetaTableItemButtonView(
+                                        label: "Unfollow",
+                                        systemImage: "person.2.slash",
+                                        action: {
+                                            send(.requestUnfollow(petname: user.petname))
+                                        },
+                                        role: .destructive
+                                    )
+                                } else {
+                                    MetaTableItemButtonView(
+                                        label: "Follow",
+                                        systemImage: "person.badge.plus",
+                                        action: {
+                                            send(.requestFollow(did: user.did))
+                                        }
+                                    )
+                                }
+                                
+                                MetaTableItemButtonView(
+                                    label: "Block",
+                                    systemImage: "hand.raised",
+                                    action: {},
+                                    role: .destructive
+                                )
+                            }
+                            
+                        }
+                        
+                        if Config.default.userProfileDetailsTable {
+                            if state.isDetailsTablePresented {
```

Comment:
`&&`?

Reply:
These can't actually be combined without losing functionality, there are three different cases here:

- Detail Section Hidden
- Detail Section Visible, Show Details Button Visible
- Detail Section Expanded, Hide Details Button Visible